src/__tests__/components/forms/CreateEmailTemplate/CreateEmailTemplate.test.tsx

```diff
@@ -1,11 +1,109 @@
+import { render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { mockDeep } from 'jest-mock-extended';
+import { TemplateFormState } from '@utils/types';
 import { CreateEmailTemplate } from '@forms/CreateEmailTemplate/CreateEmailTemplate';
-import { render, screen } from '@testing-library/react';
-
-describe('CreateEmailTemplate component', () => {
-  it('should render', () => {
-    render(<CreateEmailTemplate />);
-    expect(screen.getByTestId('page-sub-heading')).toHaveTextContent(
-      'Placeholder email page'
-    );
-  });
+
+jest.mock('@utils/amplify-utils', () => ({
+  getAmplifyBackendClient: () => {},
+}));
+
+jest.mock('react-dom', () => {
+  const originalModule = jest.requireActual('react-dom');
+
+  return {
+    ...originalModule,
+    useFormState: (
+      _: (
+        formState: TemplateFormState,
+        formData: FormData
+      ) => Promise<TemplateFormState>,
+      initialState: TemplateFormState
+    ) => [initialState, '/action'],
+  };
+});
+
+test('renders page', async () => {
+  const user = userEvent.setup();
+
+  const container = render(
+    <CreateEmailTemplate
+      initialState={mockDeep<TemplateFormState>({
+        validationError: undefined,
+        emailTemplateName: undefined,
+        emailTemplateSubjectLine: undefined,
+        emailTemplateMessage: undefined,
+      })}
+    />
+  );
+  expect(container.asFragment()).toMatchSnapshot();
+
+  const templateNameInput = await container.findByTestId(
+    'emailTemplateName-input'
+  );
+  await user.type(templateNameInput, 'template-name');
+
+  const templateSubjectLineInput = await container.findByTestId(
+    'emailTemplateSubjectLine-input'
+  );
+  await user.type(templateSubjectLineInput, 'template-subject-line');
+
+  const templateMessageInput = await container.findByTestId(
+    'emailTemplateMessage-input'
+  );
+  await user.type(templateMessageInput, 'template-message');
```

Comment:
why type in the fields and then not check anything?

Reply:
Copy and paste of NHS app tests. Will remove 👍